plugins/modules/medialive_channel_placement_group.py

```diff
@@ -0,0 +1,523 @@
+      - When I(state=present), wait for the placement group to reach the ACTIVE state.
+      - When I(state=absent), wait for the placement group to be deleted.
+    type: bool
+    required: false
+    default: true
+  wait_timeout:
+    description:
+      - The maximum time in seconds to wait for the channel placement group to reach the desired state.
+      - Defaults to 60 seconds.
+    type: int
+    required: false
+    default: 60
+
+extends_documentation_fragment:
+  - amazon.aws.common.modules
+  - amazon.aws.region.modules
+  - amazon.aws.boto3
+"""
+
+EXAMPLES = r"""
+# Create a MediaLive Channel Placement Group
+- community.aws.medialive_channel_placement_group:
+    cluster_id: '123456'
+    nodes:
+      - '34598743'
+    state: present
+
+# Update a MediaLive Channel Placement Group
+- community.aws.medialive_channel_placement_group:
+    id: '234523'
+    cluster_id: '123456'
+    name: 'UpdatedPlacementGroup'
+    nodes:
+      - '34598743'
+    state: present
+
+# Delete a MediaLive Channel Placement Group
+- community.aws.medialive_channel_placement_group:
+    id: '234523'
+    cluster_id: '123456'
+    state: absent
+"""
+
+RETURN = r"""
+placement_group:
+  description: The details of the channel placement group
+  returned: success
+  type: dict
+  contains:
+    arn:
+      description: The ARN of the channel placement group
+      type: str
+      returned: success
+      example: "arn:aws:medialive:us-east-1:123456789012:channelplacementgroup/cpg-12345"
+    channel_placement_group_id:
+      description: The ID of the channel placement group
+      type: str
+      returned: success
+      example: "234523"
+    cluster_id:
+      description: The ID of the cluster
+      type: str
+      returned: success
+      example: "123456"
+    created:
+      description: When the channel placement group was created
+      type: str
+      returned: success
+      example: "2025-03-31T23:08:55Z"
+    state:
+      description: The state of the channel placement group
+      type: str
+      returned: success
+      example: "ACTIVE"
+"""
+
+import uuid
+from typing import Dict, Literal
+
+try:
+    from botocore.exceptions import WaiterError, ClientError, BotoCoreError
+except ImportError:
+    pass # caught by AnsibleAWSModule
+
+from ansible.module_utils.common.dict_transformations import snake_dict_to_camel_dict, camel_dict_to_snake_dict, recursive_diff
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.botocore import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.exceptions import AnsibleAWSError
+from ansible_collections.community.aws.plugins.module_utils.base import BaseWaiterFactory
+
+
+class MediaLiveWaiterFactory(BaseWaiterFactory):
+    '''Custom waiter factory for MediaLive channel placement group resources'''
+
+    @property
```

Comment:
Can you please move this custom waiter to https://github.com/ansible-collections/amazon.aws/blob/main/plugins/module_utils/waiters.py?